tests/test_auth_dict.py

```diff
@@ -21,13 +21,15 @@ def tmp_home(tmpdir):
 
 
 def test_auth_from_dict():
-    """get_auth_dict, if given a dict, returns the input."""
+    """get_auth_dict, if given a dict, returns a shallow copy of the input."""
```

Comment:
The test doesn't verify that it's a shallow copy.  You should probably either not include this in the test doc string, or else actually verify that (I guess by verifying that == is true and "is" is False?)

I'd suggest the former, as the higher-level test is good enough.